### ISSUE TITLE
Rule proposal: name-dialect

### ISSUE THREAD

**Issue by Richienb** [sindresorhus/eslint-plugin-unicorn#2520]
### Description

Enforce consistent dialect (Commonweath/British or American) for function and variable names

### Fail

When `['error', 'american']`:

```js
const myColour = 'red'';
```

When `['error', 'commonwealth']`:

```js
const myColor = 'red'';
```

### Pass

When `['error', 'american']`:

```js
const myColor = 'red'';
```

When `['error', 'commonwealth']`:

```js
const myColour = 'red'';
```

### Proposed rule name

name-dialect

**Comment by fisker** [comment]
I like the idea, but may it can be done by spellcheck?

**Comment by sindresorhus** [comment]
I think a spellcheck will be too annoying for variable names. It will have too many false-positives and words it doesn't know. It's easier to manually define a list of common spelling inconsistencies. We could reuse the machinery from `prevent-abbreviations`.

**Comment by fregante** [comment]
From a "support" point of view, this should probably be added if there's a short and complete list of such words, otherwise you'll end up having to tweak it regularly via "feature requests" issues.

**Comment by silverwind** [comment]
> I like the idea, but may it can be done by spellcheck?

Most spellcheckers would probably have both variants pass. Also, many of them are a pain to set up because they tend to have too many false-positives. So I do support this proposal, I assume the list will be pretty short.

**Comment by sindresorhus** [comment]
This seems already possible with `prevent-abbreviations`, as it supports arbitrary identifier replacements, not just abbreviations. I don’t think we should add a separate dialect rule unless there’s a strong case for a curated built-in default list. A full dialect checker would be too broad and spellchecker-like, and a small list can live in user config. It could also becmoe a maintenance chore.
